.github/workflows/scripts/slack_notifications/gem_notifier.rb

```diff
@@ -0,0 +1,86 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+# This file runs on a 24 hour cycle via gem_notifications.yml and sends Slack updates for new gem version releases.
+
+require 'time'
+require 'httparty'
+require_relative 'slack_notifier'
+
+class GemNotifier < SlackNotifier
+  def self.check_for_updates(watched_gems)
+    return if verify_gem_list(watched_gems)
+    watched_gems.each do |gem_name|
+      gem_info = verify_gem(gem_name)
+      abort("Failed to obtain info for gem '#{gem_name}'") unless gem_info
+
+      versions = gem_versions(gem_info)
+      send_slack_message(gem_message(gem_name, versions)) if gem_updated?(versions)
+    end
+  end
+
+  private
+
+  def self.verify_gem_list(watched_gems)
+    abort("Nothing to see here! The 'watched_gems' array cannot be empty") if watched_gems.empty?
+  end
+
+  def self.verify_gem(gem_name)
+    gem_info = HTTParty.get("https://rubygems.org/api/v1/versions/#{gem_name}.json")
+
+    gem_info if gem_info.success?
+  end
+
+  # Gem entries are ordered by release date. The break limits versions to two versions: newest and previous.
+  def self.gem_versions(gem_info)
+    versions = gem_info.each_with_object([]) do |gem, arr|
+      next unless gem['platform'] == 'ruby'
+
+      # for the "new" version (first one recorded), report any and all types of
+      #   versions (stable, preview, rc, beta, etc.)
+      # for the "previous" version, record only the newest stable version
+      if arr.size == 0 || !gem['number'].match?(/(?:rc|beta|preview)/)
+        arr << gem
+      end
+
+      break arr if arr.size == 2
+    end
+  end
+
+  def self.gem_updated?(versions)
+    Time.now.utc - Time.parse(versions[0]['created_at']) < CYCLE
+  end
+
+  def self.github_diff(gem_name, newest, previous)
+    diff = HTTParty.get(interpolate_github_url(gem_name, newest, previous))
+
+    diff.success?
+  end
+
+  def self.interpolate_github_url(gem_name, newest, previous)
+    "https://github.com/#{gem_name}/#{gem_name}/compare/v#{previous}...v#{newest}"
```

Comment:
I think I've brought this up in other reviews, but the source code link may have the full GitHub URL, potentially providing better results for those gems that don't have the same user/org name as the gem, however, you would probably need to use a different API endpoint than `versions` to get that value.

Reply:
I really liked this comment and have made getting that `diff` more reliable! Here's the [new code](https://github.com/newrelic/newrelic-ruby-agent/pull/1432/files#diff-ddcde05fb3aba0b9770eddeb6be0d8e2a76ac7f86d69786141db8cb712412794R57-R65).